### PR TITLE
4210 deleted user team

### PR DESCRIPTION
**Pull request by sjson421**
Issue #4210

Also deletes user from team when a manager deletes that user

### CODE REVIEW

File: src/app/users/users.component.ts [hunk 1]
```diff
@@ -220,8 +222,21 @@ export class UsersComponent implements OnInit, OnDestroy, AfterViewInit {
     event.stopPropagation();
   }
 
+  deleteUserFromTeams(user) {
+    // //REPLACE THIS!
+    // this.couchService.findAll('teams').then(teamsDb => {
+    //   const docsWithUser = teamsDb.filter(doc => doc.userId === user._id)
+    //   this.couchService.bulkDocs('teams', docsWithUser.map(doc =>
+    //     this.couchService.delete('teams/' + doc._id + '?rev=' + doc._rev))
```
Comment (paulbert) [inline src/app/users/users.component.ts]: `{ ...doc, _deleted: true }`

File: src/app/users/users.component.ts [hunk 2]
```diff
@@ -220,8 +222,21 @@ export class UsersComponent implements OnInit, OnDestroy, AfterViewInit {
...
+    //   this.couchService.bulkDocs('teams', docsWithUser.map(doc =>
+    //     this.couchService.delete('teams/' + doc._id + '?rev=' + doc._rev))
+    //   );
+    // })
+  }
+
   deleteUser(user) {
     const userId = 'org.couchdb.user:' + user.name;
+
+    this.deleteUserFromTeams(user);
```
Comment (paulbert) [inline src/app/users/users.component.ts]: This line should be run as another element of the array in `forkJoin` on line 243

File: src/app/users/users.component.ts [hunk 3]
```diff
@@ -220,8 +222,21 @@ export class UsersComponent implements OnInit, OnDestroy, AfterViewInit {
     event.stopPropagation();
   }
 
+  deleteUserFromTeams(user) {
+    // //REPLACE THIS!
+    // this.couchService.findAll('teams').then(teamsDb => {
```
Comment (paulbert) [inline src/app/users/users.component.ts]: 1. Return an open observable to be used in the deleteUser function
2. You can use a selector with findAll so you don't have to filter
3. Use a switchMap to run the bulkDocs for deleting
```
return this.couchService.findAll('teams', { selector: { userId: user._id }).pipe(switchMap(teams => {
  return this.couchService.bulkDocs...
})
```

**Comment by sjson421** [comment]
@paulbert Thanks for the very detailed help

**Pull request merged**